File: leetcode/001-Two-Sum/go/Solution.go
func twoSum(nums []int, target int) []int {
numIdx := make(map[int]int)
for i, num := range nums {
if _, ok := numIdx[target - num]; ok {
return []int{numIdx[target - num], i}
}
numIdx[num] = i
}
return nil
}

File: leetcode/001-Two-Sum/kotlin/Solution.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        var numIdx: HashMap<Int, Int> = HashMap<Int, Int>()
        nums.forEachIndexed { i, num ->
            if (numIdx.containsKey(target - num)) {
                return intArrayOf(numIdx.getValue(target - num), i)
            }
            numIdx.put(num, i)
        }
        return intArrayOf()
    }
}